src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -109,13 +109,35 @@ public void jobLogic() {
             List<FeedVersion> latestVersionsWithCriticalErrors = new LinkedList<>();
             boolean shouldWaitForNewFeedVersions = false;
 
+            // Production ready feed versions.
+            List<Deployment.SummarizedFeedVersion> pinnedFeedVersions = deployment.retrievePinnedFeedVersions();
+
             // Iterate through each feed version for deployment.
             for (
                 Deployment.SummarizedFeedVersion currentDeploymentFeedVersion : deployment.retrieveFeedVersions()
             ) {
-                // Retrieve and update to the latest feed version associated with the feed source of the current
+                // Retrieve the latest feed version associated with the feed source of the current
                 // feed version set for the deployment.
                 FeedVersion latestFeedVersion = currentDeploymentFeedVersion.feedSource.retrieveLatest();
+                // Make sure the latest feed version is not going to supersede a pinned feed version.
+                if (pinnedFeedVersions
+                    .stream()
+                    .anyMatch(feedVersion -> feedVersion.feedSource.id.equals(latestFeedVersion.feedSourceId))
+                ) {
+                    String message = String.format("Feed version %s from feed source %s could not be advanced because a previous feed version has been pinned for project %s.",
+                        latestFeedVersion.id,
+                        latestFeedVersion.parentFeedSource().name,
+                        project.name);
+                    LOG.warn(message);
+                    NotifyUsersForSubscriptionJob.createNotification(
```

Comment:
I don't think this notification is needed any longer -- the email notification should only go out if the auto deploy job exits because of some unexpected or failure condition. In this case, we're just skipping the latest feed version and proceeding with the pinned version right?

Reply:
That is correct. I have removed the notification.